Makefile: add a rule to build a static library

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
[Fabrice: update for 2.43]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
diff -durN liblinear-1.96.orig/Makefile liblinear-1.96/Makefile
--- liblinear-1.96.orig/Makefile	2014-11-15 07:50:23.000000000 +0100
+++ liblinear-1.96/Makefile	2014-12-14 00:23:00.135893956 +0100
@@ -16,6 +16,11 @@
 	fi; \
 	$(CXX) $${SHARED_LIB_FLAG} linear.o newton.o blas/blas.a -o liblinear.so.$(SHVER)
 
+# Keeping blas/blas.a as a pre-requisite, to ensure all .o files are built
+static-lib: linear.o newton.o blas/blas.a
+	$(AR) rcv liblinear.a linear.o newton.o blas/*.o
+	$(RANLIB) liblinear.a
+
 train: newton.o linear.o train.c blas/blas.a
 	$(CXX) $(CFLAGS) -o train train.c newton.o linear.o $(LIBS)
 
